mitiq/zne/inference.py

```diff
@@ -422,67 +426,38 @@ def __init__(
         shot_list: Optional[List[int]] = None,
     ) -> None:
         """Instantiates a new object of this Factory class."""
-
         if order > len(scale_factors) - 1:
             raise ValueError(
                 "The extrapolation order cannot exceed len(scale_factors) - 1."
             )
         self.order = order
         super(PolyFactory, self).__init__(scale_factors, shot_list)
 
-    @staticmethod
-    def static_reduce(
-        instack: List[dict], exp_values: List[float], order: int
-    ) -> float:
+    def reduce(self) -> float:
+        """Returns the zero-noise limit found by fitting a polynomial of degree
+        `self.order` to the input data of scale factors and expectation values.
+
+        Stores the optimal parameters for the fit in `self.opt_params`.
         """
-        Determines with a least squared method, the polynomial of degree equal
-        to 'order' which optimally fits the input data.
-        The zero-noise limit is returned.
+        scale_factors = self.get_scale_factors()
+        expectation_values = self.get_expectation_values()
 
-        This static method is equivalent to the "self.reduce" method of
-        PolyFactory, but can be called also by other factories which are
-        particular cases of PolyFactory, e.g., LinearFactory
-        and RichardsonFactory.
+        if len(scale_factors) < 2:
+            raise ValueError(
+                "At least two data points are needed to fit a polynomial."
+            )
```

Comment:
This check seems unnecessary since we already check it at initialization. Given that it is missing in `LinearFactory` and in `RichardsonFactory` I would remove it also from here (`PolyFactory`) for consistency. 
```suggestion
```